rootfs/etc/nginx/lua/monitor.lua

```diff
@@ -46,15 +49,14 @@ local function flush(premature)
     return
   end
 
-  local current_metrics_batch = metrics_batch
```

Comment:
You are changing the behaviour here. The idea was to take current batch as soon as timer calls `flush` function and reset the main queue so that it can accept more metrics and continue with encoding and sending the current metrics batch while the other one is being filled.

With your change we will potentially lose more metrics. Imagine timer (flush function) kicks in right before the time when the current batch is full. It'll start encoding the metrics and during this time every new metric will be dropped.   Given cjson is quite fast at encoding there'll probably not many metrics being dropped but if we can avoid that why not.

Reply:
got it. I will fix that.

Reply:
@ElvinEfendi fixed. 
used `table.clone` to clone table: https://github.com/openresty/luajit2/blob/v2.1-agentzh/README#L65